### ISSUE TITLE
Probably bug with center crop / resize 

### ISSUE THREAD

**Issue by recoilme** [huggingface/diffusers#5830]
### Describe the bug

Diffusers apply resize before center crop and this lead to loss of most part of the picture. So if you train on not square dataset you lost parts of images during distillation

Affected code

Center crop/resize bug in:

wuerstchen/text_to_image
https://github.com/huggingface/diffusers/blob/a042909c836c794a34508b314b3ce8ce93a96284/examples/wuerstchen/text_to_image/train_text_to_image_prior.py#L658

train_text_to_image_sdxl.py
https://github.com/huggingface/diffusers/blob/a042909c836c794a34508b314b3ce8ce93a96284/examples/text_to_image/train_text_to_image_sdxl.py#L816

train_text_to_image_lora_sdxl
https://github.com/huggingface/diffusers/blob/a042909c836c794a34508b314b3ce8ce93a96284/examples/text_to_image/train_text_to_image_lora_sdxl.py#L824

train_text_to_image_lora
https://github.com/huggingface/diffusers/blob/a042909c836c794a34508b314b3ce8ce93a96284/examples/text_to_image/train_text_to_image_lora.py#L597

train_text_to_image_flax
https://github.com/huggingface/diffusers/blob/a042909c836c794a34508b314b3ce8ce93a96284/examples/text_to_image/train_text_to_image_flax.py#L332

train_text_to_image
https://github.com/huggingface/diffusers/blob/a042909c836c794a34508b314b3ce8ce93a96284/examples/text_to_image/train_text_to_image.py#L744C27-L744C27

### Reproduction

import torch
import torchvision.transforms as transforms
from PIL import Image
import matplotlib.pyplot as plt

image_path = "rec.png"
image = Image.open(image_path)

transform1 = transforms.Compose([
    transforms.CenterCrop((256)),
    transforms.Resize((256), interpolation=transforms.InterpolationMode.BILINEAR)
])
transform2 = transforms.Compose([
    transforms.Resize((256), interpolation=transforms.InterpolationMode.BILINEAR),
    transforms.CenterCrop((256))
])
tensor_image1 = transforms.ToTensor()(transform1(image))
tensor_image2 = transforms.ToTensor()(transform2(image))

fig, ax = plt.subplots(1, 2)

ax[0].imshow(tensor_image1.permute(1, 2, 0))
ax[0].axis('off')
ax[1].imshow(tensor_image2.permute(1, 2, 0))
ax[1].axis('off')
plt.tight_layout()
plt.show()

<img width="638" alt="Снимок экрана 2023-11-17 в 00 35 51" src="https://github.com/huggingface/diffusers/assets/417177/8ef4721f-cfc9-4474-92e2-e5b362e3bc31">

### System Info

latest

### Who can help?

@sayakpaul

**Comment by recoilme** [comment]
tldr; 
as is: 
**Resize, Crop** 
to be: 
**Crop, Resize**

**Comment by recoilme** [comment]
oh, everything is ok now, where did I see the opposite? Maybe I was in some old thread... got confused, sorry, I'll come back if I find this place..
